tasks/main.yml

```diff
@@ -112,22 +126,31 @@
       when: consul_config is defined
 
   no_log: true
-  when: consul_raw_key is not defined and bootstrap_marker.stat.exists and (consul_node_role == "bootstrap" or consul_node_role == "server")
+  when:
+    - consul_raw_key is not defined
+    - bootstrap_marker.stat.exists
+    - (consul_node_role == "bootstrap" or consul_node_role == "server")
 
 - name: Write key locally to share with new servers
   local_action: copy content="{{ consul_raw_key }}" dest=/tmp/consul_raw.key
   become: no
-  when: consul_raw_key is defined and bootstrap_marker.stat.exists
+  when:
+    - consul_raw_key is defined
+    - bootstrap_marker.stat.exists
 
 - name: Read key for servers that require it
   set_fact:
     consul_raw_key: "{{ lookup('file', '/tmp/consul_raw.key') }}"
-  when: consul_raw_key is not defined and bootstrap_marker.stat.exists
+  when:
+    - consul_raw_key is not defined
+    - not bootstrap_marker.stat.exists
```

Comment:
Added the `not` here to fix the bug.

Should have done it in a separate commit, but thought too late about it.